### PR TITLE
[master] Remove unnecessary closure environments for lambda functions 

### PR DESCRIPTION
**Pull request by gimantha**
…ve compiledJarFile

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/DocumentContext.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ class DocumentContext {
     private NodeCloner nodeCloner;
     private final DocumentId documentId;
     private final String name;
-    private final String content;
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/DocumentContext.java:61]: IIRC, the design here is to make everything immutable, Even though this fixes the OOM issue, not sure the design here is correct. @sameerajayasoma @azinneera

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java
Comment (Nadeeshan96) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:1]: Here https://github.com/ballerina-platform/ballerina-lang/pull/41400/files at line 6225, a clone is not created.

Comment (gimantha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:1]: Resolved

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Pull request merged**